business/metrics.go

```diff
@@ -251,39 +252,62 @@ func createStatsMetricsLabelsBuilder(q *models.MetricsStatsQuery) *MetricsLabels
 	return lb
 }
 
-func (in *MetricsService) GetControlPlaneMetrics(q models.IstioMetricsQuery, scaler func(n string) float64) (models.MetricsMap, error) {
-	metrics := make(models.MetricsMap)
+func (in *MetricsService) GetControlPlaneMetrics(q models.IstioMetricsQuery, pods models.Pods, scaler func(n string) float64) (models.MetricsMap, error) {
+	podRegex := ""
+	separator := ""
+	for _, pod := range pods {
+		podRegex = fmt.Sprintf("%s%s%s", podRegex, separator, pod.Name)
+		separator = "|"
+	}
+	podLabel := fmt.Sprintf(`{pod="%s"}`, podRegex)
 
-	h := in.prom.FetchHistogramRange("pilot_proxy_convergence_time", "", "", &q.RangeQuery)
+	metrics := make(models.MetricsMap)
 	var err error
-	converted, err := models.ConvertHistogram("pilot_proxy_convergence_time", h, models.ConversionParams{Scale: 1})
+
+	// pilot_proxy_convergence_time is handled in a special way.  our typical "sum(range(" queries for avg and quantiles,
+	// don't work here. Because proxy sync is reported more like a step function.  The count and sum are updated one time,
+	// and so there is typically no range, because any range requires at least two data points.  What we really want here
+	// is delta(sum) / delta(count) for the time period. Note, this is pretty non-standard manipulation of a histogram,
+	// don't try this at home.
+	deltaDuration := q.End.Sub(q.Start)
+	deltaSumMetric := in.prom.FetchDelta("pilot_proxy_convergence_time_sum", podLabel, "", q.End, deltaDuration)
+	deltaSumConverted, err := models.ConvertMetric("pilot_proxy_convergence_time_sum", deltaSumMetric, models.ConversionParams{Scale: 1})
+	if err != nil {
+		return nil, err
+	}
+	deltaSum := deltaSumConverted[0].Datapoints[0].Value
```

Comment:
These are always going to have 1 value?

Reply:
That's correct.  It's the delta for the counter for the time period. Basically val(tEnd) - val(tStart).